src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -66,184 +60,113 @@ enum DruidSpells
     SPELL_DRUID_STAMPEDE_CAT_STATE          = 109881
 };
 
-// 1850 - Dash
-class spell_dru_dash : public SpellScriptLoader
+//  8921 - Moonfire
+class spell_dru_moonfire : public SpellScriptLoader
 {
     public:
-        spell_dru_dash() : SpellScriptLoader("spell_dru_dash") { }
+        spell_dru_moonfire() : SpellScriptLoader("spell_dru_moonfire") { }
 
-        class spell_dru_dash_AuraScript : public AuraScript
+        class spell_dru_moonfire_SpellScript : public SpellScript
         {
-            PrepareAuraScript(spell_dru_dash_AuraScript);
+            PrepareSpellScript(spell_dru_moonfire_SpellScript);
 
-            void CalculateAmount(AuraEffect const* /*aurEff*/, int32& amount, bool& /*canBeRecalculated*/)
+            void HandleOnHit(SpellEffIndex /*effIndex*/)
             {
-                // do not set speed if not in cat form
-                if (GetUnitOwner()->GetShapeshiftForm() != FORM_CAT_FORM)
-                    amount = 0;
+                Unit* Caster = GetCaster();
+                Unit* Target = GetHitUnit();
+
+                if (Target == nullptr)
+                    return;
+
+                if (!Caster || Caster->GetTypeId() != TYPEID_PLAYER)
+                    return;
+
+                if (Unit* Target = GetHitUnit())
```

Comment:
You already declared Target.. and checked it on nullptr.. why do you checking it again?